### DISCUSSION TITLE
Configure the percent of disk usage which triggers image garbage collection

### DISCUSSION THREAD

**Discussion by Leviathan321** [k3s-io/k3s#6999] [Q&A] [answered]
I have the problem that k3s removes my images from registry after some time, even if I have more than 40gb space left.

I have seen this can be done using "kubelet" flags, but I am not familiar how to do this via k3s. ([information found](https://konubinix.eu/braindump/posts/why_does_k3d_k3s_kubernetes_clean_my_images/))

Is anyone familiar with this?

Thank you

**Comment by brandond** [comment, marked as answer]
You're looking for the image garbage collection settings. See:
https://github.com/k3s-io/k3s/discussions/6360#discussioncomment-3984096

**Reply by Leviathan321** [reply]
Thank you for the hint! I tried to set the config in the k3s.yaml file, but there is no effect. After a restart the configuration file is overwritten also. Is there anyhting I am doing wrong?

**Reply by brandond** [reply]
k3s.yaml is the admin kubeconfig for use by kubectl and other CLI tools, not the k3s server configuration. Did you mean to edit config.yaml, or the CLI flags in the k3s systemd unit?

**Comment by Leviathan321** [comment]
So what I mean is the file that is modified in the original answer
/etc/rancher/k3s/config.yaml.
I dont have a config.yaml there, just k3s. yaml. I thought I need to modify
this file then.

Brad Davidson ***@***.***> schrieb am Mo., 27. Feb. 2023,
21:47:

**Reply by brandond** [reply]
No, as I said in my initial response that's not the correct file. You can create the config file and add the desired entries, or modify the systemd unit and add the equivalent CLI flags. It's up to you which one you prefer.
